component/telemetry.go

```diff
@@ -37,13 +37,13 @@ type TelemetrySettings struct {
 
 	// ReportComponentStatus allows a component to report runtime changes in status. The service
 	// will automatically report status for a component during startup and shutdown. Components can
-	// use this method to report status after start and before shutdown. ReportComponentStatus
-	// will only return errors if the API used incorrectly. The two scenarios where an error will
-	// be returned are:
-	//
-	//   - An illegal state transition
-	//   - Calling this method before component startup
-	//
-	// If the API is being used properly, these errors are safe to ignore.
+	// use this method to report status after start and before shutdown.
+	// Deprecated: [v0.92.0] This function will be removed in a future release.
+	// Use ReportStatus instead.
 	ReportComponentStatus func(*StatusEvent) error
+
+	// ReportStatus allows a component to report runtime changes in status. The service
+	// will automatically report status for a component during startup and shutdown. Components can
+	// use this method to report status after start and before shutdown.
+	ReportStatus func(*StatusEvent)
```

Comment:
What about we use interface instead, something like:

```golang
type StatusReporter interface {
   Report(*StatusEvent)
}

type TelemetrySettings struct {
  ....
  StatusReporter StatusReporter // I am not sure about the name of the variable, I like the interface name
}
```

This allows us to add more funcs to this interface in the future. For that we can even use optional interfaces that the StatusReporters can optionally implement, or we add an un-exported func and we have a "NopStatusReporters" that everyone must embed to implement the interface.

Reply:
trying it

Reply:
a bit more research needed, especially as we can use this interface to resolve this comment:
https://github.com/open-telemetry/opentelemetry-collector/blob/b3d04f58abda0f0550b2bfa547390ee2ce150be8/component/telemetry.go#L17-L19

Reply:
I don't anticipate needing additional status reporting methods for components. If we have some examples, I'd be open to considering it. As this is a method on `component.TelemetrySettings`, to be used by components, is there any value in having a optional interface? FWIW, the service does need multiple methods, and when I made that change, I intentionally did not expose the additional method on `component.TelemetrySettings`. The other difference with the `servicetelemetry.TelemetrySettings` version is that it takes the `component.InstanceID` as a parameter, whereas the component version is bound to an `InstanceID` for a specific component via a closure.